tests/migration/eviction_strategy.go

```diff
@@ -0,0 +1,720 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package migration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	kvpointer "kubevirt.io/kubevirt/pkg/pointer"
+
+	"kubevirt.io/kubevirt/tests/testsuite"
+
+	"kubevirt.io/kubevirt/tests/clientcmd"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+	"kubevirt.io/kubevirt/tests/framework/cleanup"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libnet"
+	"kubevirt.io/kubevirt/tests/libnode"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/util"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	k8sv1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"kubevirt.io/kubevirt/tests/libwait"
+
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+)
+
+var _ = SIGMigrationDescribe("Live Migration", func() {
+	var (
+		virtClient kubecli.KubevirtClient
+		err        error
+	)
+
+	BeforeEach(func() {
+		checks.SkipIfMigrationIsNotPossible()
+		virtClient = kubevirt.Client()
+	})
+
+	setControlPlaneUnschedulable := func(mode bool) {
+		controlPlaneNodes, err := virtClient.
+			CoreV1().
+			Nodes().
+			List(context.Background(),
+				metav1.ListOptions{LabelSelector: `node-role.kubernetes.io/control-plane`})
+		Expect(err).ShouldNot(HaveOccurred(), "could not list control-plane nodes")
+		Expect(controlPlaneNodes.Items).ShouldNot(BeEmpty(),
+			"There are no control-plane nodes in the cluster")
+
+		for _, node := range controlPlaneNodes.Items {
+			if node.Spec.Unschedulable == mode {
+				continue
+			}
+
+			nodeCopy := node.DeepCopy()
+			nodeCopy.Spec.Unschedulable = mode
+
+			oldData, err := json.Marshal(node)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			newData, err := json.Marshal(nodeCopy)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			patch, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, node)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			_, err = virtClient.CoreV1().Nodes().Patch(context.Background(), node.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+		}
+	}
+
+	// temporaryNodeDrain also sets the `NoSchedule` taint on the node.
+	// nodes with this taint will be reset to their original state on each
+	// test teardown by the test framework. Check `libnode.CleanNodes`.
+	temporaryNodeDrain := func(nodeName string) {
```

Comment:
nit: also `libinfra`

Reply:
I tried to move this to the `libnode` package, but the `make` fails due to a circular dependency.
Basically, moving this function, something like this comes into being:
```
//  .-> //tests/testsuite:go_default_library
//  |   //tests/libnode:go_default_library
//  |   //tests/clientcmd:go_default_library
//  `-- //tests/testsuite:go_default_library
```
To resolve this cycle, probably we should move the `testsuite.GetTestNamespace` function into a higher level.
Do the latter in this PR could create a lot of `rebase` cycles, since `testsuite.GetTestNamespace` function is widely used everywhere!
I wrote a `TODO` above the function as reminder.
Please, let me know. Thanks

Reply:
SGTM, thanks!